contracts/SpokePool.sol

```diff
@@ -147,19 +179,88 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     }
 
     function initiateRelay(
-        uint256 originChain,
-        address sender,
+        uint256 originChainId,
         uint256 amount,
-        address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
+        uint64 depositId,
+        uint64 relayerFeePct,
+        uint64 realizedLpFeePct,
+        address relayToken,
+        address sender,
+        address recipient
+    ) public {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18 && realizedLpFeePct <= 0.5e18, "invalid fees");
+
+        // Use relay count as unique ID for new relay.
+        require(relays[numberOfRelays].relayAmount == 0, "Relay exists");
+        relays[numberOfRelays] =
+            RelayData(
+                recipient,
+                relayToken,
```

Comment:
can we rename this variable to `relayDestinationToken` to make it clear it's the token on the destination?

Reply:
sure